src/main/java/seedu/address/logic/commands/BatchEditCommand.java

```diff
@@ -0,0 +1,116 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.PersonContainsTagsPredicate;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Edit all person with specified tag to change the specified tag with new tag.
+ */
+public class BatchEditCommand extends Command {
+    public static final String COMMAND_WORD = "batch-edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Change the specified tag with new tag"
+            + "Parameters:"
+            + PREFIX_TAG + "TAG" + " "
+            + PREFIX_TAG + "TAG\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_TAG + "OLDTAG " + PREFIX_TAG + "NEWTAG";
+
+    public static final String MESSAGE_BATCH_EDIT_EACH_PERSON_SUCCESS = "Tag changed from: %1$s\n";
```

Comment:
Works great just that the result message shows the old tag before editing

Reply:
i have added change to section.

Reply:
Oki can but inputting a wrong tag shows an empty message. Similiar to edit and delete commands etc i think we could have an error message like tag specified does not exist or smth

Reply:
Yupp LGTM now.